operator/apis/machinelearning/v1/seldondeployment_types.go

```diff
@@ -72,15 +74,27 @@ func GetExplainerDeploymentName(sdepName string, predictorSpec *PredictorSpec) s
 	}
 }
 
-func GetDeploymentName(mlDep *SeldonDeployment, predictorSpec PredictorSpec, podSpec *SeldonPodSpec, idx int) string {
-	name := mlDep.Name + "-" + predictorSpec.Name
+func getContainerNames(containers []v1.Container) string {
+	name := ""
+	for i, c := range containers {
+		if i > 0 {
+			name = name + "-"
+		}
+		name = name + c.Name
+	}
+	return name
+}
+
+func GetDeploymentName(mlDep *SeldonDeployment, predictorSpec PredictorSpec, podSpec *SeldonPodSpec, podSpecIdx int) string {
+	baseName := mlDep.Name + "-" + predictorSpec.Name + "-" + strconv.Itoa(podSpecIdx) + "-"
+	var name string
 	if podSpec != nil && len(podSpec.Metadata.Name) != 0 {
-		name = name + "-" + podSpec.Metadata.Name
+		name = baseName + podSpec.Metadata.Name
```

Comment:
If `podSpec` has a name defined, but the containers change name after an update, wouldn't we run into the same issue we are trying to solve?

Perhaps we could always append the hashed list of container names?

Reply:
If you specify the name in the podSpec meta then you are taking control of whether its a new or updated deployment that is created.